spyder/plugins/editor/widgets/tests/test_bookmarks.py

```diff
@@ -0,0 +1,275 @@
+def reset_emits(editor):
+    "Reset signal mocks."
+    editor.bookmarks_changed.reset_mock()
+
+
+def editor_assert_helper(editor, block=None, bm=None, emits=True):
+    """Run the tests for call to add_remove_breakpoint.
+
+    Args:
+        editor: CodeEditor instance.
+        block: Block of text.
+        bm: A list containing slots and columns of bookmarks
+        emits: Boolean to test if signals were emitted?
+    """
+    data = block.userData()
+    assert data.bookmarks == bm
+    if emits:
+        editor.bookmarks_changed.emit.assert_called_with()
+    else:
+        editor.bookmarks_changed.emit.assert_not_called()
+
+
+# --- Fixtures
+# -----------------------------------------------------------------------------
+@pytest.fixture
+def code_editor_bot(qtbot):
+    """Create code editor with default Python code."""
+    editor = codeeditor.CodeEditor(parent=None)
+    indent_chars = ' ' * 4
+    tab_stop_width_spaces = 4
+    editor.setup_editor(language='Python', indent_chars=indent_chars,
+                        tab_stop_width_spaces=tab_stop_width_spaces)
+    # Mock the signal emit to test when it's been called.
+    editor.bookmarks_changed = Mock()
+    text = ('def f1(a, b):\n'
+            '"Double quote string."\n'
+            '\n'  # Blank line.
+            '    c = a * b\n'
+            '    return c\n'
+            )
+    editor.set_text(text)
+    return editor, qtbot
+
+
+# ---- Qt Test Fixtures
+@pytest.fixture
+def setup_editor(qtbot, monkeypatch):
+    """Set up the Editor plugin."""
+    qapplication()
+    from spyder.plugins.editor.plugin import Editor
+
+    monkeypatch.setattr('spyder.plugins.editor.plugin.add_actions', Mock())
+
+    class MainMock(QMainWindow):
+        def __getattr__(self, attr):
+            if attr.endswith('actions'):
+                return []
+            elif attr == 'projects':
+                projects = Mock()
+                projects.get_active_project.return_value = None
+                return projects
+            else:
+                return Mock()
+
+    window = MainMock()
+    editor = Editor(window)
+    window.setCentralWidget(editor)
+    window.resize(640, 480)
+    qtbot.addWidget(window)
+    window.show()
+
+    yield editor
+
+
+@pytest.fixture(scope="module")
+def test_file(tmpdir_factory):
+    """Create and save some python code and text in temporary file."""
+    tmpdir = tmpdir_factory.mktemp("files")
+
+    filename = osp.join(tmpdir.strpath, 'foo1.py')
+    with open(filename, 'w') as f:
+        f.write("# -*- coding: utf-8 -*-\n"
+                "def foo:\n"
+                "    print(Hello World!)\n")
+
+    return filename
+
+
+# --- Tests
+# -----------------------------------------------------------------------------
+def test_add_bookmark(code_editor_bot, mocker):
+    """Test CodeEditor.add_bookmark. Adds bookmark data to Textblock."""
+    editor, _ = code_editor_bot
+    ab = editor.add_bookmark
```

Comment:
Is there a particular justification for remapping this name? It just seems to obfuscate the logic of the test merely to save a few characters, which are nearly inconsequential if using autocompletion or line duplication.

Reply:
Nope, no logic, removed.